### DISCUSSION TITLE
Stock: Using resources - a very easy and simple solution

### DISCUSSION THREAD

**Discussion by mehmetkuzu** [bounswe/bounswe2023group2#65] [Ideas]
In any system covering flow of things there exists a "stock" entity.

I will try to show by example:

We define resources. 
So a resource is something promised by the giver. And we aim to connect this resource to a need.
Let's say

Mr. Çamlıbel has 100 kgs of rice in his office in İstanbul ready to give away (Resource ID: 10901)
Let's assume Mr. Çamlıbel calls us and says "You have to take these. I dont have place to put them temporarily.

Now arises a stock problem. Physical problem is not our issue. But how will be represent this in our design.
Will we introduce stock points and stock entries?

No need.
We simply convert the resource by Mr. Çamlıbel to a resource by our system (or the voluntary depot owner, who took the rice from Çamlıbel's place)

So this is the flow:
- A resource is declared by the user (Çamlıbel Probably)
- User informs us about the urgency (should be taken away as soon as possible)

Then comes the flow:
- An action is input to the system
-- Action uses (consumes) the resource by Çamlıbel
-- Creates a new resource by the Depot (with the goods, instrumenets etc. in the first resource record)

Done.

So that also shows that:
- A resource is actually a stock for some cases.
- As any stock entity resources can change place/owner etc.

**Comment by cahidenes** [comment]
Seems reasonable. This also shows that an action shall be able to handle resource transportation.

Is there any further action we should take about this discussion? If not, we can close the discussion.

**Comment by mehmetkuzu** [comment]
I think one of the "action templates" we talked about yesterday covers this.
Put to stock - like action -> One resource (which will be transfered) in - one resource in the stock place out.
So we can close the discussion.
